### PR TITLE
Update collector to v0.104.0

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?
This PR updates all collector dependencies to v0.104.0 except datadogconnector. datadogconnector cannot be upgraded because of breaking change in #26347.

### Describe how to test/QA your changes
Spin up agent with OTLP ingest with defaults and make sure the http/ grpc servers addresses are `localhost` instead of `0.0.0.0`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1349,6 +1350,9 @@ exclude (
 	github.com/knadh/koanf/providers/confmap v0.1.0
 )
 
+// todo: update datadog connector with breaking changes from https://github.com/DataDog/datadog-agent/pull/26347.
```
Comment (songy23) [inline go.mod:1353]: cc @ajgajg1134 with the `replace` this PR shouldn't block  https://github.com/DataDog/datadog-agent/pull/26347, but please let us know if you ran into dependency issues.

**Review by songy23 — COMMENTED** [review]
As suggested by @mx-psi please add a release note

**Review by dineshg13 — CHANGES_REQUESTED** [review]
Please don't edit comp/otelcol/collector-contrib/impl/go.mod directly. Instead update in 
`comp/otelcol/collector-contrib/impl/manifest.yaml` and run `inv collector.generate` . 

Also please update the version here https://github.com/DataDog/datadog-agent/blob/d8bdcaf6208f1d3d36d5b8a4ae1dc9d65924b64e/comp/otelcol/collector/impl/collector.go#L94

**Comment by dineshg13** [comment]
We do need a QA to ensure the upstream breaking changes aren't impacting OTLP ingest in the Agent

**Comment by mackjmr** [comment]
@dineshg13 I updated `manifest.yaml` and version in `collector.yaml`. I also added QA instructions, although I don't think there is anything to check other than the http/ grpc servers addresses are `localhost` instead of `0.0.0.0`.

**Review by dineshg13 — APPROVED** [review]
We are using datadog connector `0.104.0` , but otherwise LGTM

**Review by hestonhoffman — APPROVED** [review]

**Comment by songy23** [comment]
> We are using datadog connector 0.104.0 , but otherwise LGTM

It is replaced with 0.103.0 so shouldn't be a problem

**Review by songy23 — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]

File: releasenotes/notes/collector-104-8697c1d5cc4b1745.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Change default bind address in OTLP ingest from `0.0.0.0` to `localhost`. Please refer to the following blog post: https://opentelemetry.io/blog/2024/hardening-the-collector-one for additional information around this change.
```
Comment (GustavoCaso) [inline releasenotes/notes/collector-104-8697c1d5cc4b1745.yaml:11]: The release note has nothing to do with the title of the PR, is that intended?

File: comp/otelcol/collector-contrib/def/go.mod [hunk 3]
```diff
@@ -34,32 +34,26 @@ require (
 	github.com/prometheus/procfs v0.15.0 // indirect
 	github.com/shirou/gopsutil/v4 v4.24.5 // indirect
 	github.com/shoenig/go-m1cpu v0.1.6 // indirect
-	github.com/spf13/cobra v1.8.0 // indirect
+	github.com/spf13/cobra v1.8.1 // indirect
```
Comment (GustavoCaso) [inline comp/otelcol/collector-contrib/def/go.mod:37]: Is this bump intended?

File: releasenotes/notes/collector-104-8697c1d5cc4b1745.yaml (hunk 2 above)
Comment (mackjmr) [reply releasenotes/notes/collector-104-8697c1d5cc4b1745.yaml:11]: The bump to v0.104.0 brings in the change mentioned in the changelog, see: https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.104.0.

Comment (GustavoCaso) [reply releasenotes/notes/collector-104-8697c1d5cc4b1745.yaml:11]: Nit: Maybe mention the upgrade to `0.104.0` of the collector

**Review by GustavoCaso — APPROVED** [review]

**Comment by mackjmr** [comment]
> @dineshg13 @songy23
We are using datadog connector 0.104.0 , but otherwise LGTM
Are you talking about the version in manifest.yaml ? I can bump this down to 0.103.0 ?

I didn't realize `inv collector.generate` was removing the replace statement in `comp/otelcol/collector-contrib/impl/go.mod`. I've updated the datadogconnector in manifest.yaml to v0.103.0 with a todo. I've also updated the OCB version to v0.104.0 in manifest.yaml.

**Review by FlorentClarret — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM for Agent DevX Infra file !

**Review by hghotra — APPROVED** [review]
👍🏼 - Serverless

**Pull request merged**